src/app/pages/all-meals/all-meals.component.ts

```diff
@@ -0,0 +1,93 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Meal } from 'src/app/models/meal';
+import { MealData } from 'src/app/models/mealBody.model';
+import { MenuService } from 'src/app/services/menu.service';
+
+@Component({
+  selector: 'app-all-meals',
+  templateUrl: './all-meals.component.html',
+  styleUrls: ['./all-meals.component.scss'],
+})
+export class AllMealsComponent implements OnInit {
+  meals: Meal;
+  message = false;
+  letters = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z'];
+  //Sorting
+  key: string;
+  reverse: boolean = false;
+  // Pagination
+  p: number = 1;
+  // Search by
+  searchKey: string = '';
+  sarchTerm: string = '';
+  // Search change by name or ingredients
+  ingredients = false;
+
+  constructor(private menuService: MenuService, private router: Router, private route: ActivatedRoute) {}
+
+  ngOnInit(): void {
+    // Initialize list of meals by first letter
+    this.menuService.listAllMealsByFirstLetter('a').subscribe((meals: Meal) => {
```

Comment:
why call service again when you have a method doing this bellow, be consistent, that way when you need to change something you change it in fewer places